src/extension/app/plugins/delete/delete.js

```diff
@@ -0,0 +1,104 @@
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { log } from '../../../log.js';
+import { SidekickPlugin } from '../../components/plugin/plugin.js';
+import {
+  MODALS, MODAL_EVENTS, TOAST_EVENTS, RESTRICTED_PATHS,
+} from '../../constants.js';
+
+/**
+ * @typedef {import('@AppStore').AppStore} AppStore
+ */
+
+/**
+ * @typedef {import('@Types').CorePlugin} CorePlugin
+ */
+
+/**
+ * Creates the delete plugin
+ * @param {AppStore} appStore The app store
+ * @returns {SidekickPlugin} The delete plugin
+ */
+export function createDeletePlugin(appStore) {
+  return new SidekickPlugin({
+    id: 'delete',
+    condition: (store) => store.isPreview()
+      && store.status.code?.status !== 200 // // only show if not code
+      && !RESTRICTED_PATHS.includes(store.location.pathname), // or restricted path
+    // pinned: false, // TODO: set to unpinned
+    button: {
+      text: appStore.i18n('delete'),
+      action: async () => {
+        const { location, status } = appStore;
+        const isPage = status.webPath.split('/').pop().indexOf('.') === -1;
+        const hasSrc = status.edit?.status === 200;
+
+        // double check
+        if (hasSrc && !appStore.isAuthenticated()) {
+          const message = isPage
+            ? appStore.i18n('delete_page_source_exists')
+            : appStore.i18n('delete_file_source_exists');
+          appStore.showToast(message, 'negative');
+          return;
+        }
+
+        // get user confirmation
+        const message = isPage
+          ? appStore.i18n(hasSrc ? 'delete_page_confirm' : 'delete_page_no_source_confirm')
+          : appStore.i18n(hasSrc ? 'delete_file_confirm' : 'delete_file_no_source_confirm');
+        const modal = appStore.showModal({
+          type: MODALS.DELETE,
+          data: {
+            headline: appStore.i18n('delete'),
+            message,
+            confirmLabel: appStore.i18n('delete'),
+          },
+        });
+        modal.addEventListener(MODAL_EVENTS.CONFIRM, async () => {
+          // perform delete
+          appStore.showWait();
+          try {
+            const resp = await appStore.delete();
+            appStore.hideWait();
+            if (resp.ok) {
+              // show success toast
+              const toast = appStore.showToast(
+                isPage
+                  ? appStore.i18n('delete_page_success')
+                  : appStore.i18n('delete_file_success'),
+                'positive',
+              );
+              toast.addEventListener(TOAST_EVENTS.CLOSE, () => {
```

Comment:
Feel like this is going to seem odd waiting 6 seconds after the deletion is complete to redirect. Let's see though, maybe it's not an issue.